rules/integrations/aws/persistence_ec2_instance_request_to_iam_service.toml

```diff
@@ -0,0 +1,81 @@
+[metadata]
+creation_date = "2024/07/24"
+integration = ["aws"]
+maturity = "production"
+updated_date = "2024/07/24"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies when an EC2 instance interacts with the AWS IAM service via an assumed role. This is uncommon behavior and could indicate an attacker using compromised creadentials to further exploit an environment. For example, an assumed role could be used to create new users for persistence or add permissions for privilege escalation. An EC2 instance assumes a role using their EC2 id as the session name. This rule looks for the pattern "i-" which is the beginning pattern for assumed role sessions started by an EC2 instance.
+"""
+false_positives = ["Administrators may use EC2 instances to interact with IAM services as part of an automation workflow, ensure validity of the triggered event and include exceptions where necessary."]
+from = "now-6m"
+index = ["filebeat-*", "logs-aws.cloudtrail-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "AWS EC2 Instance Interaction with IAM Service"
+references = [
+    "https://redcanary.com/blog/aws-sts/",
+]
+risk_score = 47
+rule_id = "a44bcb58-5109-4870-a7c6-11f5fe7dd4b1"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Data Source: AWS",
+    "Data Source: Amazon Web Services",
+    "Data Source: AWS EC2",
+    "Data Source: AWS IAM",
+    "Use Case: Identity and Access Audit",
+    "Tactic: Privilege Escalation",
+    "Tactic: Persistence",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+any where event.dataset == "aws.cloudtrail"
```

Comment:
@imays11 is there any indication in the data that the source of this API request is from an EC2 instance? Would be nice to have entity store to add this context :)

Maybe the source IP is in a common EC2 CIDR or the ARN of the source indicates EC2?

Reply:
Yes, the indicator that this is an EC2 instance is the instance id attached to the end of the `user.id` field and `aws.cloudtrail.user_identity.arn` which starts with `i-` You can see that detection method described [here](https://catalog.workshops.aws/aws-cirt-ssrf-imdsv1-simulation-and-detection/en-US/detect/detect02) by the AWS CIRT team